### PR TITLE
[HOPSWORKS-1770] reindex featurestore index

### PR DESCRIPTION
**Pull request by o-alex**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -133,6 +135,8 @@ int main(int argc, char** argv) {
          "Table tailer barrier type. EPOCH=0, GCI=1")
         ("reindex", po::value<bool>(&reindex)->default_value(reindex),
          "initialize an empty index with all metadata")
+        ("reindex_of", po::value<std::string>(&reindex_of)->default_value(reindex_of),
+         "define index to initialize")
```
Comment (maismail) [inline src/main.cpp:139]: write in the description the possible values [project, featurestore]

File: src/main.cpp [hunk 2]
```diff
@@ -212,15 +216,22 @@ int main(int argc, char** argv) {
...
+                                             config, elastic_index, elastic_batch_size,
+                                             elastic_issue_time, lru_cap);
+        reindexer->run();
+      } else if(reindex_of == "featurestore") {
+        LOG_INFO("Create Elasticsearch index at " << elastic_featurestore_index);
+        FeaturestoreReindexer *reindexer = new FeaturestoreReindexer(connection_string.c_str(),
+                database_name.c_str(), meta_database_name.c_str(), hive_meta_database_name.c_str(), config,
+                elastic_featurestore_index, elastic_batch_size, elastic_issue_time, lru_cap);
+        reindexer->run();
+      }
```
Comment (maismail) [inline src/main.cpp:234]: what about both? reindexing featurestore and project, maybe we should another condition for All?

File: src/main.cpp [hunk 3]
```diff
@@ -55,6 +56,7 @@ int main(int argc, char** argv) {
     Barrier barrier = EPOCH;
 
     bool reindex = false;
+    std::string reindex_of = "project";
```
Comment (maismail) [inline src/main.cpp:59]: you need to add this attribute to the config.ini.template 
and also we need to propagate this information from elasticsearch-chef and epipe-chef
https://github.com/logicalclocks/elasticsearch-chef/blob/master/attributes/default.rb#L48
https://github.com/logicalclocks/epipe-chef/blob/master/recipes/default.rb#L67

File: src/main.cpp (hunk 2 above)
Comment (o-alex) [reply src/main.cpp:234]: it is more complicated and adds little benefit... you could just first run project reindex and then featurestore reindex.

File: src/main.cpp (hunk 1 above)
Comment (o-alex) [reply src/main.cpp]: fixed

File: src/main.cpp (hunk 2 above)
Comment (o-alex) [reply src/main.cpp:245]: fixed

File: src/main.cpp (hunk 3 above)
Comment (o-alex) [reply src/main.cpp:59]: fixed

File: src/main.cpp [hunk 4]
```diff
@@ -41,7 +42,7 @@ int main(int argc, char** argv) {
     TableUnitConf provenance_tu = TableUnitConf();
 
     bool hopsworks = true;
-    std::string elastic_index = "projects";
+    std::string elastic_index = "all";
```
Comment (maismail) [inline src/main.cpp:45]: i guess this is a typo, i guess you meant to change reindex_of to all

Comment (o-alex) [reply src/main.cpp]: fixed... sry... typo indeed

**Comment by maismail** [comment]
@o-alex change the commit message to "[HOPSWORKS-1770] reindex featurestore index"

**Pull request merged**